### PR TITLE
Add `mermaid` rendering

### LINKED ISSUE #567: Render mermaid
- add mermaid rendering feature in markdown
- cf.
    - https://github.com/raghur/mermaid-filter
    - [using pandoc filters to create graphs with hakyll](https://enter-haken.github.io/posts/2017-02-20-pandoc-filter.html)

### PR DESCRIPTION
**Pull request by falgon**
- close #567 
- Embed the svg using mermaid-cli
- Currently, it is only valid for articles and resume pages.
    - To enable it on the home page, do it in the home page markdown generation, related #566 
    - This feature will be available once mermaid-cli v11 is released, so it still fails for now.
        - 👀 https://github.com/mermaid-js/mermaid-cli/pull/701

### CODE REVIEW

**Comment by kiirotori** [comment]
* [CircleCI](https://dl.circleci.com/status-badge/redirect/gh/falgon/roki%2Dweb/tree/add%2Dmermaid%2Drendering)
* [Build log](https://circleci.com/gh/falgon/roki-web/1023)
* [Preview build](https://output.circle-artifacts.com/output/job/1240f801-e43e-4056-a66e-35b218142e67/artifacts/0/index.html) with this PR (696bb3ba8d83609d574c26ee6195770627bae540)
<hr /><details><summary>more info</summary><p><ul><li>:cloud: Storage Free space   7.88 GiB    52.56%</li><li>:open_file_folder: 148.29 MiB  CI_WORK/GitHubActions/roki-web</li></ul></p></details>

**Comment by kiirotori** [comment]
* [CircleCI](https://dl.circleci.com/status-badge/redirect/gh/falgon/roki%2Dweb/tree/add%2Dmermaid%2Drendering)
* [Build log](https://circleci.com/gh/falgon/roki-web/1024)
* [Preview build](https://output.circle-artifacts.com/output/job/2b611de5-d8c3-4a12-9d27-fbf35790892d/artifacts/0/index.html) with this PR (c433afb4d2e3a441d30492d461e046fd16c2e258)
<hr /><details><summary>more info</summary><p><ul><li>:cloud: Storage Free space   7.71 GiB    51.43%</li><li>:open_file_folder: 148.29 MiB  CI_WORK/GitHubActions/roki-web</li></ul></p></details>

**Comment by kiirotori** [comment]
* [CircleCI](https://dl.circleci.com/status-badge/redirect/gh/falgon/roki%2Dweb/tree/add%2Dmermaid%2Drendering)
* [Build log](https://circleci.com/gh/falgon/roki-web/1027)
* [Preview build](https://output.circle-artifacts.com/output/job/bc6ae2a1-d148-4aa9-a837-04571939bb02/artifacts/0/index.html) with this PR (40cd29195caeda89f760f394ceed59d3593376c3)
<hr /><details><summary>more info</summary><p><ul><li>:cloud: Storage Free space   7.20 GiB    48.03%</li><li>:open_file_folder: 148.29 MiB  CI_WORK/GitHubActions/roki-web</li></ul></p></details>

**Comment by kiirotori** [comment]
* [CircleCI](https://dl.circleci.com/status-badge/redirect/gh/falgon/roki%2Dweb/tree/add%2Dmermaid%2Drendering)
* [Build log](https://circleci.com/gh/falgon/roki-web/1028)
* [Preview build](https://output.circle-artifacts.com/output/job/749d4a09-3ebc-40de-8cf8-b6e3c4a2649a/artifacts/0/index.html) with this PR (71950e5bd5f647f5edb6a214d04e020abf406b4f)
<hr /><details><summary>more info</summary><p><ul><li>:cloud: Storage Free space   7.03 GiB    46.89%</li><li>:open_file_folder: 148.29 MiB  CI_WORK/GitHubActions/roki-web</li></ul></p></details>

File: src/Media/SVG.hs [hunk 1]
```diff
@@ -1,9 +1,53 @@
...
+
+execMmdc :: (MonadIO m, Monad n, MonadThrow m) => T.Text -> m (HtmlT n ())
+execMmdc = liftIO . readCreateProcessWithExitCode (proc "npx" args) . T.unpack >=> \case
+    (ExitFailure _, _, err) -> throwString err
+    (ExitSuccess, out, _)   -> pure $ toHtmlRaw $ T.pack out
+    where
+        args = ["mmdc", "-i", "/dev/stdin", "-e", "svg", "-o", "-"]
+
+mermaidCodeBlock :: Block -> Compiler Block
+mermaidCodeBlock cb@(CodeBlock (_, _, t) contents) = maybe cb id <$>
```
Comment (falgon) [inline src/Media/SVG.hs:38]:
```suggestion
mermaidCodeBlock cb@(CodeBlock (_, _, t) contents) = fromMaybe cb <$>
```

File: .github/workflows/build_pr.yml [hunk 2]
```diff
@@ -53,11 +53,24 @@ jobs:
         ghc-version: ${{ env.ghc_version }}
         enable-stack: true
         stack-version: ${{ env.stack_version }}
-    - name: Install tools
+    - name: Install tools (apt)
       uses: awalsh128/cache-apt-pkgs-action@v1.4.2
       with:
-        packages: libxml2-utils parallel
+        packages: libxml2-utils parallel chromium-browser
```
Comment (falgon) [inline .github/workflows/build_pr.yml:59]:
```suggestion
        packages: libxml2-utils parallel
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -49,11 +49,24 @@ jobs:
       with:
         path: ~/.stack
         key: stack-v2-${{ runner.os }}-${{ hashFiles('stack.yaml') }}-${{ hashFiles('stack.yaml.lock') }}-${{ hashFiles('package.yaml') }}-${{ hashFiles('docker/.env') }}
-    - name: Install tools
+    - name: Install tools (apt)
       uses: awalsh128/cache-apt-pkgs-action@v1.4.2
       with:
-        packages: libxml2-utils
+        packages: libxml2-utils chromium-browser
```
Comment (falgon) [inline .github/workflows/build.yml:55]:
```suggestion
        packages: libxml2-utils
```

**Comment by kiirotori** [comment]
* [CircleCI](https://dl.circleci.com/status-badge/redirect/gh/falgon/roki%2Dweb/tree/add%2Dmermaid%2Drendering)
* [Build log](https://circleci.com/gh/falgon/roki-web/1029)
* [Preview build](https://output.circle-artifacts.com/output/job/24349346-bbf0-4393-9233-676db59058ba/artifacts/0/index.html) with this PR (0c275ede9750069184a2aa2dc2608c535615f8e4)
<hr /><details><summary>more info</summary><p><ul><li>:cloud: Storage Free space   6.86 GiB    45.76%</li><li>:open_file_folder: 148.29 MiB  CI_WORK/GitHubActions/roki-web</li></ul></p></details>

**Comment by kiirotori** [comment]
* [CircleCI](https://dl.circleci.com/status-badge/redirect/gh/falgon/roki%2Dweb/tree/add%2Dmermaid%2Drendering)
* [Build log](https://circleci.com/gh/falgon/roki-web/1030)
* [Preview build](https://output.circle-artifacts.com/output/job/3de19310-57bd-4a4a-bad2-6d31d25a51e3/artifacts/0/index.html) with this PR (d20a365613699ceaa66eed32890cd8f03ad69247)
<hr /><details><summary>more info</summary><p><ul><li>:cloud: Storage Free space   6.69 GiB    44.62%</li><li>:open_file_folder: 148.29 MiB  CI_WORK/GitHubActions/roki-web</li></ul></p></details>

**Comment by kiirotori** [comment]
* [CircleCI](https://dl.circleci.com/status-badge/redirect/gh/falgon/roki%2Dweb/tree/add%2Dmermaid%2Drendering)
* [Build log](https://circleci.com/gh/falgon/roki-web/1031)
* [Preview build](https://output.circle-artifacts.com/output/job/f59b0684-2dbb-4c3e-ad21-9a4844caf893/artifacts/0/index.html) with this PR (d87fc5a6e8b895ec0f9f6306653bce07990af800)
<hr /><details><summary>more info</summary><p><ul><li>:cloud: Storage Free space   6.52 GiB    43.48%</li><li>:open_file_folder: 148.29 MiB  CI_WORK/GitHubActions/roki-web</li></ul></p></details>

**Comment by kiirotori** [comment]
* [CircleCI](https://dl.circleci.com/status-badge/redirect/gh/falgon/roki%2Dweb/tree/add%2Dmermaid%2Drendering)
* [Build log](https://circleci.com/gh/falgon/roki-web/1032)
* [Preview build](https://output.circle-artifacts.com/output/job/6f98034d-16ea-4109-a5f3-162298733efd/artifacts/0/index.html) with this PR (c1badde2d07cca3cb29dcb83914cbe820c7e6a47)
<hr /><details><summary>more info</summary><p><ul><li>:cloud: Storage Free space   6.35 GiB    42.35%</li><li>:open_file_folder: 148.29 MiB  CI_WORK/GitHubActions/roki-web</li></ul></p></details>

**Comment by kiirotori** [comment]
* [CircleCI](https://dl.circleci.com/status-badge/redirect/gh/falgon/roki%2Dweb/tree/add%2Dmermaid%2Drendering)
* [Build log](https://circleci.com/gh/falgon/roki-web/1033)
* [Preview build](https://output.circle-artifacts.com/output/job/bc1fe7b7-6357-49ce-ba1e-99a63d280604/artifacts/0/index.html) with this PR (c1badde2d07cca3cb29dcb83914cbe820c7e6a47)
<hr /><details><summary>more info</summary><p><ul><li>:cloud: Storage Free space   6.18 GiB    41.22%</li><li>:open_file_folder: 148.29 MiB  CI_WORK/GitHubActions/roki-web</li></ul></p></details>

**Review by kiirotori — APPROVED** [review]

**Pull request merged**